STM32/ACTenChannel/Core/Src/ACTenChannel.c

```diff
@@ -0,0 +1,395 @@
+/*
+ * ACTenChannel.c
+ */
+
+#include <math.h>
+#include <string.h>
+#include <stdio.h>
+#include <ctype.h>
+#include <stdarg.h>
+#include <stdbool.h>
+#include <float.h>
+#include <inttypes.h>
+
+#include "main.h"
+#include "HeatCtrl.h"
+#include "ADCMonitor.h"
+#include "systemInfo.h"
+#include "USBprint.h"
+#include "CAProtocol.h"
+#include "CAProtocolStm.h"
+#include "StmGpio.h"
+#include "ACTenChannel.h"
+#include "pcbversion.h"
+#include "DS18B20.h"
+
+/***************************************************************************************************
+** DEFINES
+***************************************************************************************************/
+
+#define ADC_CHANNELS	        10
+#define ADC_CHANNEL_BUF_SIZE	400
+
+#define MAX_TEMPERATURE         70
+
+#define USB_COMMS_TIMEOUT_MS    5000
+
+/***************************************************************************************************
+** PRIVATE TYPEDEFS
+***************************************************************************************************/
+
+typedef struct ActuationInfo {
+    int pin; // pins 0-9 are interpreted as single ports - pin '-1' is interpreted as all
+    int pwmDutyCycle;
+    int timeOn; // time on is in seconds
+    bool isInputValid;
+} ActuationInfo;
+
+/***************************************************************************************************
+** PRIVATE FUNCTION DECLARATIONS
+***************************************************************************************************/
+
+static void CAallOn(bool isOn, int duration);
+static void CAportState(int port, bool state, int percent, int duration);
+static void userInput(const char *input);
+static void printAcTenChannelStatus();
+static void updateBoardStatus();
+
+/***************************************************************************************************
+** PRIVATE OBJECTS
+***************************************************************************************************/
+
+/* GPIO settings. */
+StmGpio heaterPorts[AC_TEN_CH_NUM_PORTS];
+static StmGpio fanCtrl;
+static StmGpio DQ1;
```

Comment:
What does DQ mean?

Reply:
DQ is the data line of the DS18B20 chip according to the datasheet :)